File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private List<Integer> kthSmallestList = new ArrayList<>();
    public void helper(TreeNode root){
        if(root == null){
            return;
        }
        helper(root.left);
        kthSmallestList.add(root.val);
        helper(root.right);
    }
    public int kthSmallest(TreeNode root, int k) {
        helper(root);
        return kthSmallestList.get(k-1);
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def printQueue(self, q: List[int]) -> List[int]:
        copy = []
        for i in range(len(q)):
            copy.append(q[i].val)
        return copy
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        queue = deque()
        res = []

        if root:
            queue.append(root)
            res.append(root.val)
        
        while len(queue) > 0:
            for _ in range(len(queue)):
                curr = queue.popleft()
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            if len(queue) > 1:
                res.append(queue[1].val)
            elif queue:
                res.append(queue[0].val)
                
        return res